timely/src/lib.rs

```diff
@@ -108,8 +108,8 @@ impl<T: Clone+'static> Data for T { }
 ///
 /// The `ExchangeData` trait extends `Data` with any requirements imposed by the `timely_communication`
 /// `Data` trait, which describes requirements for communication along channels.
-pub trait ExchangeData: Data + encoding::Data { }
-impl<T: Data + encoding::Data> ExchangeData for T { }
+pub trait ExchangeData: Data + encoding::Data + columnar::Columnar { }
+impl<T: Data + encoding::Data + columnar::Columnar> ExchangeData for T { }
```

Comment:
This .. worked out ok but I realized is from a prior attempt that wanted to columnarize things. Unnecessary for this PR.